setup.py

```diff
@@ -160,6 +160,7 @@ def main():
             "-DBUILD_PERF_TESTS=OFF",
             "-DBUILD_DOCS=OFF",
             "-DPYTHON3_LIMITED_API=ON",
+            "-DBUILD_OPENEXR=ON",
```

Comment:
Added this flag, because after merging https://github.com/opencv/opencv/pull/21329 macOS builds are used [`openexr`](https://github.com/opencv/opencv-python/runs/4632020184?check_suite_focus=true#step:4:2327) and [`zlib`](https://github.com/opencv/opencv-python/runs/4632020184?check_suite_focus=true#step:4:2321) from the local environment, which could break the build in the end ([error log](https://github.com/opencv/opencv-python/runs/4632020184?check_suite_focus=true#step:4:3530)).